src/Components/Components/src/Reflection/PropertySetter.cs

```diff
@@ -17,6 +17,10 @@ internal sealed class PropertySetter
         "ReflectionAnalysis",
         "IL2060:MakeGenericMethod",
         Justification = "The referenced methods don't have any DynamicallyAccessedMembers annotations. See https://github.com/mono/linker/issues/1727")]
+    [SuppressMessage(
+        "AOT",
+        "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.",
+        Justification = "The referenced methods are AOT safe with reference types.")]
```

Comment:
I don't think this suppression is valid. We won't ever call this class with a property of type `int`, `double`, or `Guid`, or any other Value Type?

Instead, if we want to make this AOT-compatible, we will need to do something like this:

https://github.com/dotnet/runtime/blob/66b11d7f2c2b25fc3e6685b02f016ef2f9f8c64f/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs#L1378-L1391

Basically - check if `!RuntimeFeature.IsDynamicCodeSupported`. When it isn't, and one of the types is a ValueType, then fallback to "slower" reflection based property setting that boxes.